### PR TITLE
WIP: Make fomrobject_t an opaque type

### PR DESCRIPTION
**Pull request by youngar**
Make `fomrobject_t` an opaque type and move towards supporting runtime compressed references in the GC example.

### CODE REVIEW

File: example/glue/MarkingDelegate.hpp [hunk 1]
```diff
@@ -173,7 +173,7 @@ class MM_MarkingDelegate
 	getObjectScanner(MM_EnvironmentBase *env, omrobjectptr_t objectPtr, void *scannerSpace, MM_MarkingSchemeScanReason reason, uintptr_t *sizeToDo)
 	{
 		GC_MixedObjectScanner *objectScanner = GC_MixedObjectScanner::newInstance(env, objectPtr, scannerSpace, 0);
-		*sizeToDo = sizeof(fomrobject_t) + objectScanner->getBytesRemaining();
+		*sizeToDo = sizeof(ObjectHeader) + objectScanner->getBytesRemaining();
```
Comment (rwy0717) [inline example/glue/MarkingDelegate.hpp:176]:
```suggestion
		*sizeToDo = objectPtr->sizeInBytes();
```

File: example/glue/MixedObjectScanner.hpp [hunk 2]
```diff
@@ -47,7 +49,7 @@ class GC_MixedObjectScanner : public GC_ObjectScanner
 	 * @param[in] flags Scanning context flags
 	 */
 	MMINLINE GC_MixedObjectScanner(MM_EnvironmentBase *env, omrobjectptr_t objectPtr, uintptr_t flags)
-		: GC_ObjectScanner(env, (fomrobject_t *)objectPtr + 1, 0, flags)
+		: GC_ObjectScanner(env, (fomrobject_t *)((uintptr_t)objectPtr + sizeof(ObjectHeader)), 0, flags)
```
Comment (rwy0717) [inline example/glue/MixedObjectScanner.hpp:52]:
```suggestion
		: GC_ObjectScanner(env, (fomrobject_t *)(object->begin()), 0, flags)
```

Comment (rwy0717) [reply example/glue/MixedObjectScanner.hpp:52]: `Object::begin(env, object);`

**Comment by youngar** [comment]
This code should be working fine now.  This PR will be updated to implement the `Opaque fomrobjectptr_t` proposal as it unfolds.

@genie-omr build all

**Comment by youngar** [comment]
@genie-omr build all

**Comment by youngar** [comment]
@genie-omr build all

**Comment by youngar** [comment]
@genie-omr build all

**Comment by youngar** [comment]
@genie-omr build all

**Pull request closed without merging**